src/java/io/bazel/rulesscala/scalac/ProtoReporter.java

```diff
@@ -0,0 +1,108 @@
+package io.bazel.rulesscala.scalac;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import scala.reflect.internal.util.Position;
+import scala.reflect.internal.util.RangePosition;
+import scala.tools.nsc.Settings;
+import scala.tools.nsc.reporters.AbstractReporter;
+
+import io.bazel.rules_scala.diagnostics.Diagnostics;
+
+public class ProtoReporter extends AbstractReporter {
+
+  private final Settings settings;
+  Map<String, List<Diagnostics.Diagnostic>> builder;
+
+  public ProtoReporter(Settings settings) {
+    this.settings = settings;
+    builder = new LinkedHashMap<>();
+  }
+
+  public void writeTo(Path path) throws IOException {
+    Diagnostics.TargetDiagnostics.Builder targetDiagnostics = Diagnostics.TargetDiagnostics.newBuilder();
+    for (Map.Entry<String, List<Diagnostics.Diagnostic>> entry : builder.entrySet()) {
+      targetDiagnostics.addDiagnostics(Diagnostics.FileDiagnostics.newBuilder().setPath(entry.getKey()).addAllDiagnostics(entry.getValue()));
+    }
+    Files.write(path, targetDiagnostics.build().toByteArray(), StandardOpenOption.CREATE, StandardOpenOption.APPEND);
+  }
+
+  @Override
+  public Settings settings() {
+    return settings;
+  }
+
+  @Override
+  public void display(Position pos, String msg, Severity severity) {
+  }
+
+  @Override
+  public void displayPrompt() {
+  }
+
+  @Override
+  public void info0(Position pos, String msg, Object severity, boolean force) {
+      Diagnostics.Diagnostic diagnostic = Diagnostics.Diagnostic
+          .newBuilder()
+          .setRange(positionToRange(pos))
+          .setSeverity(convertSeverity(severity))
+          .setMessage(msg)
+          .build();
+      // TODO: Handle generated files
```

Comment:
should this be handled? also relates to tests (for existing behavior and missing)

Reply:
I'm not sure myself it would make much sense, if one can assume the quality of the generated files. Perhaps Daniel could have a better opinion on this, since he created this file.